registry/txt.go

```diff
@@ -69,7 +73,7 @@ func NewTXTRegistry(provider provider.Provider, txtPrefix, txtSuffix, ownerID st
 	cacheInterval time.Duration, txtWildcardReplacement string,
 	managedRecordTypes, excludeRecordTypes []string,
 	txtEncryptEnabled bool, txtEncryptAESKey []byte,
-	newFormatOnly bool) (*TXTRegistry, error) {
+	newFormatOnly bool, isMigrationEnabled bool, oldOwnerID string) (*TXTRegistry, error) {
```

Comment:
can we have right after the code block 
```go
	if ownerID == "" {
		return nil, errors.New("owner id cannot be empty")
	}
```

a block something like

```go
	if isMigrationEnabled && oldOwnerID == "" {
		return nil, errors.New("oldOwnerID cannot be empty if isMigrationEnabled is true")
	}
```

or something like that? This should simplify some of the logic. As well, add unit test(s) for that

Reply:
What about cases when `oldOwnerID == ownerID`, shell we exclude that case too?